spec/infinispan_cluster_spec.js

```diff
@@ -30,11 +33,467 @@ describe('Infinispan cluster client', function() {
       .catch(t.failed(done)).finally(done);
   });
 
+  it('can put a key/value pair using different clients', function(done) {
```

Comment:
These tests feel a bit repetitive compare to what is done in local spec. There's an opportunity here to refactor the local spec tests so that you can pass in two clients: the client that does the operation (e.g. put) and the client that checks the result (e.g. get), and then call it different clients. In the case of local, both the client that does the operation and checks the result is the same.

Also, I'm not sure the tests you've written here are doing what you expect them to do. What is it that you are trying to verify here to be precise? 

The reason I ask is because no matter which cluster node each client connects to, it will get the topology and it will route request to the owner node (they are talking to a distributed cache), hence all you are testing is that if a client 1 goes to node A to put a value, client 2 will go to node A to retrieve the value.

I am wondering whether you real intention was to have client 1 store data in node A and then client 2 being able to retrieve the value from node B? If that was your intention, this test is not really achieving that.